The ``0-add_integer`` module
============================

Using ``add_integer``
---------------------

Import function from module:

    >>> add_integer = __import__('0-add_integer').add_integer

Test one int argument:

    >>> add_integer(2)
    100

Test one int argument and check default second:

    >>> add_integer(0)
    98

Test one float:
 
    >>> add_integer(123.9)
    221

Test one negative float:

    >>> add_integer(-95.5)
    3

Test two integers:
   
    >>> add_integer(20, 10)
    30

Test two big integers:
    
    >>> add_integer(999999996, 999999999)
    1999999995

Test positive, negative integers:

    >>> add_integer(34263478, -89234582395)
    -89200318917

Test negative, positive integers:

    >>> add_integer(-13947692, 3568)
    -13944124

Test an integer and a float:

    >>> add_integer(123, 7.987)
    130

Test a float and an integer:

    >>> add_integer(2.0, 3)
    5

Test two floats:

    >>> add_integer(0.1, 0.3)
    0

Add infinity:

    >>> add_integer(float('inf'), float('-inf'))
    Traceback (most recent call last):
    ...
    OverflowError: cannot convert float infinity to integer

Add nan:

    >>> add_integer(0, float('nan'))
    Traceback (most recent call last):
    ...
    ValueError: cannot convert float NaN to integer

Test first arg invalid:

    >>> add_integer("bar", 0)
    Traceback (most recent call last):
        ...
    TypeError: a must be an integer

Test second arg invalid:

    >>> add_integer(12332, [])
    Traceback (most recent call last):
        ...
    TypeError: b must be an integer

Test both args invalid:

    >>> add_integer(None, None)
    Traceback (most recent call last):
        ...
    TypeError: a must be an integer

Test two strings:

    >>> add_integer("solo", "learn")
    Traceback (most recent call last):
        ...
    TypeError: a must be an integer

Test float overflow:
    >>> add_integer(float('inf'), 0)
    Traceback (most recent call last):
    ...
    OverflowError: cannot convert float infinity to integer
